spec.bs

```diff
@@ -6,57 +6,417 @@ Status: CG-DRAFT
 Group: WICG
 URL: https://github.com/WICG/shared-storage
 Editor: Camillia Smith Barnes, Google https://google.com, cammie@chromium.org
+Markup Shorthands: markdown yes
 Abstract: Shared Storage is a storage API that is intentionally not partitioned by top-frame site (though still partitioned by context origin of course!). To limit cross-site reidentification of users, data in Shared Storage may only be read in a restricted environment that has carefully constructed output gates.
 </pre>
+<pre class=link-defaults>
+spec:url; type:interface; text:URL
+</pre>
+<style>
+/* .XXX from https://resources.whatwg.org/standard.css */
+.XXX {
+  color: #D50606;
+  background: white;
+  border: solid #D50606;
+}
+</style>
 
 Introduction {#intro}
 =====================
 <em>This section is not normative.</em>
 
-This document introduces a new primitive.
+In order to prevent cross-site user tracking, browsers are partitioning all forms of storage by top-frame site. But, there are many legitimate use cases currently relying on unpartitioned storage.
+
+This document introduces a new storage API that is intentionally not partitioned by top-frame site (though still partitioned by context origin), in order to serve a number of the use cases needing unpartitioned storage. To limit cross-site reidentification of users, data in Shared Storage may only be read in a restricted environment that has carefully constructed output gates. Over time, we hope to design and add additional gates.
```

Comment:
If I understand correctly this will require a worklet with a non-opaque origin in order to access the correct shared storage bits.  AFAIK worklets have always had an opaque origin up until now.

(I know at least related aspect of this is that worklet script requests are not treated as non-subresource requests when being intercepted by a service worker.  This was previously ok because they were opaque origin globals and therefore couldn't interact with the SW anyway.  But a non-opaque origin worklet would change this.  This may just be a purity nuisance for me and not a real problem, though.)

@domenic Are you aware of any other concerns with creating non-opaque worklets?  Why are other worklets opaque in the first place?

Reply:
Apologies for the ambiguity. The worklet will have an opaque origin.  It's the origin of the worklet's creator that I was referring to.  Clarified in the spec.